.github/actions/build-docs/action.yml

```diff
@@ -13,15 +13,31 @@ runs:
       with:
         include_grpc: 'true'
         include_dev: 'true'
+        include_asyncio: 'true'
         python_version: ${{ inputs.python-version }}
+
+    - name: Pretend this project requires Python 3.11
+      shell: bash
+      run: |
+        # Poetry won't let me install sphinx as a dev dependency in this project
+        # because of the wide range of versions our library supports. So during this
+        # action, we'll pretend this project requires Python 3.11 or greater.
+        sed -i 's/python = "^3.9"/python = "^3.11"/' pyproject.toml
+        poetry lock
+        poetry install -E grpc -E asyncio
+
+    - name: Install sphinx
+      shell: bash
+      run: |
+        poetry add sphinx myst-parser --group dev
```

Comment:
This is a minor atrocity, but for now it's the best way to work around sphinx's relatively narrow python version support. Poetry won't let me install it normally (and add it as a dev dependency in `pyproject.toml`) because our library has a wider range of python version support than `sphinx`.